psi4/src/psi4/libmints/mintshelper.cc

```diff
@@ -1996,4 +2046,1146 @@ void MintsHelper::play()
 {
 }
 
-}  // namespace psi
+/* 1st and 2nd derivatives of OEI in AO basis  */
+
+std::vector<SharedMatrix> MintsHelper::ao_overlap_kinetic_deriv1_helper(const std::string & type, int atom)
+{
+        char lbl[32];
+        char lbl1[32];
+        char ** cartcomp;
+        cartcomp = (char **) malloc (3 * sizeof(char *));
+        cartcomp[0] = strdup("X");
+        cartcomp[1] = strdup("Y");
+        cartcomp[2] = strdup("Z");
+
+        sprintf(lbl, "ao_%s_deriv1_", type.c_str());
+
+        std::shared_ptr<OneBodyAOInt> GInt;
+
+        if (type == "OVERLAP"){
+           std::shared_ptr<OneBodyAOInt> Int(integral_->ao_overlap(1));
+           GInt = Int;
+        }    
+        else {
+           std::shared_ptr<OneBodyAOInt> Int(integral_->ao_kinetic(1));
+           GInt = Int;
+        }        
+
+        std::shared_ptr <BasisSet> bs1 = GInt->basis1();
+        std::shared_ptr <BasisSet> bs2 = GInt->basis2();
+
+        int nbf1 = bs1->nbf();
+        int nbf2 = bs2->nbf();
+
+        std::vector<SharedMatrix> grad;
+        for (int p=0; p<3; p++){
+            sprintf(lbl1, "%s%d_%s", lbl, atom, cartcomp[p]);
+            grad.push_back(SharedMatrix(new Matrix(lbl1, nbf1, nbf2)));
+            }
+
+        const double* buffer = GInt->buffer();
```

Comment:
If you dont use the std library make sure to pop `cartcomp`.

Reply:
I am using the std library.